packages/cli/src/commands/update.ts

```diff
@@ -0,0 +1,136 @@
+import fetch from "node-fetch";
+import inquirer from "inquirer";
+import semver from "semver";
+import { readFileSync } from 'fs';
+import { logger } from "../utils/logger.js";
+import { getUserPkgManager } from "../utils/getUserPkgManager.js";
+import { exec } from "child_process";
+import { installDependencies } from "../utils/installDependencies.js";
+import { resolvePath } from "../utils/parseNameAndPath.js";
+import { writeJSONFile } from "../utils/fileSystem.js";
+
+export async function updateCommand(path: string) {
+    try {
+        const resolvedPath = resolvePath(path);
+        const installedPackages = await getInstalledPackages();
+        const triggerDevPackagesName = Object.keys(installedPackages.dependencies).filter(pkg => pkg.startsWith('@trigger.dev/'));
+
+        if (triggerDevPackagesName.length === 0) {
+            logger.info('No @trigger.dev packages found in package.json.');
+            return;
+        }
+
+        const triggerDevPackagesVersions = Object.fromEntries(triggerDevPackagesName.map(packageName => {
+            const currentVersion = installedPackages.dependencies[packageName].version;
+            return [packageName, currentVersion];
+        }));
+
+        const latestVersions = await getLatestVersions(triggerDevPackagesName, triggerDevPackagesVersions);
+        const updates: { [packageName: string]: string } = {};
+
+        triggerDevPackagesName.forEach(packageName => {
+            const currentVersion = installedPackages.dependencies[packageName].version;
+            const latestVersion = latestVersions[packageName];
+            if (latestVersion) {
+                if (semver.gt(latestVersion, currentVersion)) {
+                    updates[packageName] = latestVersion;
+                }
+            }
+        });
+
+
+        if (Object.keys(updates).length === 0) {
+            console.log('No updates available.');
+            return;
+        }
+
+        const confirm = await confirmUpdate(updates);
+
+        if (confirm) {
+            const updatedPackageJson = updatePackageJson(updates);
+            await writeJSONFile('package.json', updatedPackageJson);
+            logger.info('package.json updated.');
+
+            // Run 'npm install' to install the updated packages
+            logger.info('Running npm install...');
+            await installDependencies(resolvedPath);
+            logger.info('Dependencies updated.');
+        } else {
+            logger.info('Update canceled.');
+        }
+    } catch (error) {
+        logger.error('An error occurred:', error);
+    }
+}
+
+async function getInstalledPackages(): Promise<any> {
```

Comment:
This function doesn't use the resolvedPath, which means it won't be listing the correct dependencies. I also tried running this locally and it failed because `pnpm list --depth=0 --json` returned an array, not an object:

```js
[
  {
    name: 'triggerdotdev',
    version: '0.0.1',
    path: '/Users/eric/code/triggerdotdev/trigger.dev',
    private: true,
    dependencies: { '@changesets/cli': [Object], 'node-fetch': [Object] },
    devDependencies: {
      '@manypkg/cli': [Object],
      '@tailwindcss/forms': [Object],
      '@tailwindcss/typography': [Object],
      '@trigger.dev/cli': [Object],
      autoprefixer: [Object],
      'eslint-config-custom': [Object],
      postcss: [Object],
      prettier: [Object],
      tailwindcss: [Object],
      tsx: [Object],
      turbo: [Object],
      vite: [Object],
      'vite-tsconfig-paths': [Object],
      vitest: [Object]
    },
    unsavedDependencies: {
      prisma: [Object],
      '@prisma/engines': [Object],
      '@prisma/client': [Object],
      '@prisma/engines-version': [Object]
    }
  }
]
```